Sources/POSIX/stat.swift

```diff
@@ -10,14 +10,44 @@
 
 import libc
 
-/// Get file status.
-//
-// FIXME: We should probably return our own wrapper type to insulate clients
-// from platform dependencies.
-public func stat(_ path: String) throws -> libc.stat {
-    var buf = libc.stat()
-    guard libc.stat(path, &buf) == 0 else {
-        throw SystemError.dirfd(errno, path)
-    }
-    return buf
+/// Extensions to the libc `stat` structure to interpret the contents in more readable ways.
+extension libc.stat {
+
+     /// File system entity kind.
+     public enum Kind {
+         case file, directory, symlink, fifo, blockdev, chardev, socket, unknown
+
+         private init?(mode: mode_t) {
+             switch mode {
+                 case S_IFREG:  self = .file
+                 case S_IFDIR:  self = .directory
+                 case S_IFLNK:  self = .symlink
+                 case S_IFBLK:  self = .blockdev
+                 case S_IFCHR:  self = .chardev
+                 case S_IFSOCK: self = .socket
+             default:
+                 return nil
+             }
+         }
+     }
+
+     /// Kind of file system entity.
+     public var kind: Kind? {
+         return Kind(mode: st_mode & S_IFMT)
+     }
+ }
+
+/// Returns a structure containing information about the file system entity at `path`, or nil
+/// if that path doesn't exist in the file system.  Read, write or execute permission of the 
+/// file system entity at `path` itself is not required, but all ancestor directories must be searchable.
+/// If they are not, or if any other file system error occurs, this function throws a SystemError.
+/// If `followSymlink` is true and the file system entity at `path` is a symbolic link, it is traversed;
+/// otherwise it is not (any symbolic links in path components other than the last one are always traversed).
+/// If symbolic links are followed and the file system entity at `path` is a symbolic link that points to a
+/// non-existent path, then this function returns nil.
+public func stat(_ path: String, followSymlink: Bool = true) throws -> libc.stat {
```

Comment:
Agreed.  The similar code in PR #501 was slated for `Basic` so the difference in name as actually a benefit.